log-viewer/src/main/java/com/logviewer/data2/Log.java

```diff
@@ -170,6 +167,25 @@ private Path decompressAndCopyGZipFile() throws IOException {
         return tempFile;
     }
 
+    private Path decompressAndCopyZipFile() throws IOException {
+        Path tempFile = Files.createTempFile(Utils.getTempDir(), "unpacked-zip-", "-" + file.getName(file.getNameCount() - 1) + ".tmp");
```

Comment:
A zip archive may contain several files, you take the first only.
What is the usecase? Some logging system packs logs in zip instead of .gz?

Reply:
yes, exactly that.

Reply:
Just curious which is the log system?

Reply:
> Just curious which the log system packs logs in zip files?

I'd say that Spring Boot has the ability to zip old logs to a .zip